handlers/handler-fault-injection/src/main/java/org/apache/servicecomb/faultinjection/FaultInjectionHandler.java

```diff
@@ -0,0 +1,216 @@
+      int errorCode = getFaultInjectionConfig(invocation,
+          "abort.httpStatus",
+          FAULT_INJECTION_ABORT_ERROR_MSG_DEFAULT);
+      // if request need to be abort then return failure with given error code
+      CommonExceptionData errorData = new CommonExceptionData("aborted by fault inject");
+      asyncResp.consumerFail(
+          new InvocationException(new HttpStatus(errorCode, "aborted by fault inject"), errorData));
+      return;
+    }
+
+    // if no delay and no abort then continue to next handler.
+    invocation.next(asyncResp);
+  }
+
+  /**
+   * It will check the delay/abort condition based on request count and percentage
+   * received.
+   * 
+   * @param reqCount
+   * @param percentage
+   * @param key
+   * @return true/false
+   */
+  private boolean checkFaultInjectionDelayAndAbort(long reqCount, int percentage) {
+    /*
+     * Example: delay/abort percentage configured is 10% and Get the count(suppose
+     * if it is 10th request) from map and calculate resultNew(10th request) and
+     * requestOld(9th request). Like this for every request it will calculate
+     * current request count and previous count. if both not matched need to add
+     * delay/abort otherwise no need to add.
+     */
+
+    // calculate the value with current request count.
+    long resultNew = (reqCount * percentage) / 100;
+
+    // calculate the value with previous count value.
+    long resultOld = ((reqCount - 1) * percentage) / 100;
+
+    // if both are not matching then delay/abort should be added.
+    if (resultNew != resultOld) {
+      return true;
+    }
+    return false;
+  }
+
+  /**
+   * Handles the reading fault injection configuration.
+   * 
+   * @param invocation
+   *            invocation of request
+   * @param key
+   *            configuration key
+   * @return configuration value
+   */
+  private int getFaultInjectionConfig(Invocation invocation, String key, int defaultValue) {
+    int value = 0;
+    String config;
+    FaultInjectionConfig faultCfg = new FaultInjectionConfig();
+
+    //first need to check in config center map which has high priority.
+    Map<String, AtomicInteger> cfgMap = FaultInjectionUtil.getConfigCenterMap();
+
+    // get the config base on priority. operationName-->schema-->service-->global
+    String operationName = invocation.getOperationName();
+    if (operationName != null) {
+      if (invocation.getTransport().getName().equals(FAULTINJECTION_HIGHWAY_TRANSPORT)) {
+        config = CONSUMER_FAULTINJECTION_OPERATION + operationName + "."
+            + CONSUMER_FAULTINJECTION_HIGHWAY + key;
+      } else {
+        config = CONSUMER_FAULTINJECTION_OPERATION + operationName + "."
+            + CONSUMER_FAULTINJECTION_REST + key;
+      }
+
+      if (cfgMap.containsKey(config)) {
+        return cfgMap.get(config).get();
+      }
+
+      value = faultCfg.getConfigVal(config, FAULT_INJECTION_CFG_NULL);
```

Comment:
actually cfgMap we are maintaining for handling the config-center notification about config changes. if the key is present in this then it will take from this else it will get from microservice.yaml file.